### PR TITLE
[Issue 548]Fixed Carbon Core find bugs and cleaned code

### PR DESCRIPTION
**Pull request by ravipesala**
This PR fixes all 64 find bug issues. Please check the link for find bug issues
http://136.243.101.176:8080/job/CarbondataFindbugs/1/findbugsResult/module.-1658638337/

And also cleaned the unused code.

### CODE REVIEW

File: core/src/main/java/org/carbondata/core/file/manager/composite/LoadFolderData.java [hunk 1]
```diff
@@ -20,10 +20,6 @@
 package org.carbondata.core.file.manager.composite;
 
 public class LoadFolderData extends AbstractFileManager {
```
Comment (jackylk) [inline core/src/main/java/org/carbondata/core/file/manager/composite/LoadFolderData.java]: This class seems doing nothing, can we remove it?

File: core/src/main/java/org/carbondata/core/reader/ThriftReader.java [hunk 2]
```diff
@@ -89,7 +89,9 @@ public void open() throws IOException {
    * This method will set the position of stream from where data has to be read
    */
   public void setReadOffset(long bytesToSkip) throws IOException {
-    dataInputStream.skip(bytesToSkip);
+    if(dataInputStream.skip(bytesToSkip) != bytesToSkip) {
```
Comment (jackylk) [inline core/src/main/java/org/carbondata/core/reader/ThriftReader.java]: add a space after `if`

File: core/src/main/java/org/carbondata/core/writer/sortindex/CarbonDictionarySortInfoPreparator.java [hunk 3]
```diff
@@ -37,20 +36,7 @@
...
-  /**
-   * carbon table identifier instance to identify the databaseName & tableName
-   */
-  private CarbonTableIdentifier carbonTableIdentifier;
-
-  public CarbonDictionarySortInfoPreparator(String carbonStorePath,
-      CarbonTableIdentifier carbonTableIdentifier) {
-    this.carbonStorePath = carbonStorePath;
-    this.carbonTableIdentifier = carbonTableIdentifier;
+  public CarbonDictionarySortInfoPreparator() {
```
Comment (jackylk) [inline core/src/main/java/org/carbondata/core/writer/sortindex/CarbonDictionarySortInfoPreparator.java]: why to keep an empty constructor?

File: core/src/main/java/org/carbondata/query/carbon/result/iterator/MemoryBasedResultIterator.java [hunk 4]
```diff
@@ -39,8 +41,11 @@ public MemoryBasedResultIterator(Result result) {
    * @return the next element in the iteration
    */
   @Override public Result next() {
-    hasNext = false;
-    return result;
+    if(hasNext) {
```
Comment (jackylk) [inline core/src/main/java/org/carbondata/query/carbon/result/iterator/MemoryBasedResultIterator.java]: please add a space after if

File: core/src/main/java/org/carbondata/query/filter/executer/IncludeFilterExecuterImpl.java [hunk 5]
```diff
@@ -78,50 +78,48 @@ protected BitSet getFilteredIndexes(DimensionColumnDataChunk dimensionColumnData
...
-
-            for (int index = 0;
-                 index < listOfColumnarKeyBlockDataForNoDictionaryVals.size(); index++) {
-              if (ByteUtil.UnsafeComparer.INSTANCE
-                  .compareTo(filterVal, listOfColumnarKeyBlockDataForNoDictionaryVals.get(index))
-                  == 0) {
-                bitSet.set(index);
-              }
+          }
+        } else if (null != columnReverseIndexArray) {
```
Comment (jackylk) [inline core/src/main/java/org/carbondata/query/filter/executer/IncludeFilterExecuterImpl.java]: remove empty line

File: core/src/main/java/org/carbondata/query/filter/executer/IncludeFilterExecuterImpl.java [hunk 6]
```diff
@@ -78,50 +78,48 @@ protected BitSet getFilteredIndexes(DimensionColumnDataChunk dimensionColumnData
...
 
+          for (int index : columnReverseIndexArray) {
+            byte[] noDictionaryVal =
+                listOfColumnarKeyBlockDataForNoDictionaryVals.get(columnReverseIndexArray[index]);
+            if (ByteUtil.UnsafeComparer.INSTANCE.compareTo(filterVal, noDictionaryVal) == 0) {
+              bitSet.set(index);
+            }
+          }
+        } else {
+
```
Comment (jackylk) [inline core/src/main/java/org/carbondata/query/filter/executer/IncludeFilterExecuterImpl.java]: remove empty line

File: core/src/main/java/org/carbondata/query/filter/resolver/ConditionalFilterResolverImpl.java [hunk 7]
```diff
@@ -180,10 +180,9 @@ public DimColumnResolvedFilterInfo getDimColResolvedFilterInfo() {
    *
    * @return start IndexKey
```
Comment (jackylk) [inline core/src/main/java/org/carbondata/query/filter/resolver/ConditionalFilterResolverImpl.java]: please correct the javadoc for parameter and return

File: core/src/main/java/org/carbondata/query/filter/resolver/FilterResolverIntf.java [hunk 8]
```diff
@@ -71,7 +71,7 @@
    * @param startKey
    * @return long[], array of start keys.
```
Comment (jackylk) [inline core/src/main/java/org/carbondata/query/filter/resolver/FilterResolverIntf.java]: please correct the javadoc

File: core/src/main/java/org/carbondata/query/filter/resolver/RestructureFilterResolverImpl.java [hunk 9]
```diff
@@ -180,7 +180,7 @@ public DimColumnResolvedFilterInfo getDimColResolvedFilterInfo() {
    *
    * @return IndexKey.
```
Comment (jackylk) [inline core/src/main/java/org/carbondata/query/filter/resolver/RestructureFilterResolverImpl.java]: javadoc

**Comment by ravipesala** [comment]
@jackylk Fixed review comments. Please review it.

**Comment by jackylk** [comment]
Following testcase sometimes is failing in my laptop

Failed tests:   testZooKeeperLockingByTryingToAcquire2Locks(org.carbondata.core.locks.ZooKeeperLockingTest)

Tests run: 77, Failures: 1, Errors: 0, Skipped: 0

**Comment by ravipesala** [comment]
@jackylk There were three issues with that testcase
1. Some times test case hangs while starting the zookeeper server, that is the bug in zookeeper 3.4.8, it could be solved by either downgrading to 3.4.7 or overriding the ZooKeeperServer start method. Please have a look at https://issues.apache.org/jira/browse/ZOOKEEPER-2383
2. It generates very big log files under core/version-2 folder. It could be solved by moving the folder to target.
3. Test case some times failing because it cannot get the node, Need to analyze why it is happening.May we are not giving enough time to initialize the server.

I am working on it and will raise the PR. Please ignore the error as of now.

**Comment by jackylk** [comment]
Yes, we need to solve these issues. 
Thanks for working on this.

**Pull request merged**